src/Servers/Kestrel/test/InMemory.FunctionalTests/ResponseTests.cs

```diff
@@ -1426,12 +1426,15 @@ await connection.Receive(
                     $"Date: {server.Context.DateHeaderValue}",
                     "",
                     "");
+
+                connection.ShutdownSend();
+                await connection.ReceiveEnd();
             }
         }
     }
 
     [Fact]
-    public async Task HeadResponseBodyNotWrittenWithAsyncWrite()
+    public async Task HeadResponseHeadersWrittenWithAsyncWriteBeforeAppCompletes()
```

Comment:
https://github.com/aspnet/KestrelHttpServer/pull/1204 changed some of the HEAD response tests to only check the response headers are flushed and doesn't fully check that the body doesn't exist. So split the test into two for the flushed headers check, and the lack of body check.